streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1121,13 +1122,31 @@ public String name() {
 
             @Override
             public void apply(final StreamTask task) {
-                task.suspend();
+                try {
+                    task.suspend();
+                } catch (final Exception e) {
+                    try {
+                        task.close(false);
+                    } catch (final Exception f) {
+                        log.error("{} Closing task {} failed: ", logPrefix, task.id, f);
+                    }
+                    throw e;
```

Comment:
I think you are correct, we probably could carry on with the rebalance. We'd need to keep track of any active tasks that fail during suspension and remove them from the `prevActiveTasks` set that is updated in `removeStreamsTasks`
For standby tasks we'd need to remove the state directory as that is what is used to determine if a thread has any cachedTasks. Alternatively we could change it such that we keep track of the previous standby tasks in a similar way to how we do active tasks.